lib/internal/bootstrap/node.js

```diff
@@ -622,6 +622,24 @@
       // It's possible that defaultTriggerAsyncId was set for a constructor
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
+      // if node-report is enabled, call into its handler to see
+      // whether it is interested in handling the situation.
+      // Ignore if the error is scopped inside a domain.
+      if (er.domain == null) {
+        try {
+          const report = internalBinding('node_report');
+          if (report != null) {
+            if (process.binding('config').experimentalReport) {
```

Comment:
Please also emit an experimental warning (https://github.com/nodejs/node/issues/9036) using `util.emitExperimentalWarning` (https://github.com/nodejs/node/pull/16497).

Reply:
sure; the location you point out is exercised only on uncaught exceptions, so probably the ideal place is in lib/utils.js - but unfortunately that code block runs too early in the bootup phase to be able to run the `emitExperimentalWarning` (process.emitWarning is not a function etc.)

So as a reasonable workaround, I have added the warning in all the report APIs - does that sound reasonable to you?

Reply:
this still need confirmation from either @addaleax or @richardlau : the semantics (time and space) of spitting out experimental warning. Please can you confirm we are good here? thanks.